app/models/business.rb

```diff
@@ -2,15 +2,27 @@ class Business < ApplicationRecord
   has_many :users
   has_many :locations
   has_many :menus
+  has_one  :business_display_attribute
   
   before_validation :convert_business_url
 
   validates :name,         presence: true
-  validates :email,        presence: true 
-  validates :phone_number, presence: true 
-  validates :business_url, presence: true
+  validates :business_url, presence: true, uniqueness: true
```

Comment:
I validate urls params like...
`validates :url_id, presence: true, format: { with: /\A[a-z\d][a-z\d-]*[a-z\d]\z/i, message: 'only letters, numbers, and dashes.' }` 
If you use that you can remove your `convert_business_url` method.